ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -4,54 +4,163 @@
 package com.daml.platform.server.api.validation
 
 import com.daml.ledger.api.domain.LedgerId
+import com.daml.ledger.grpc.GrpcStatuses
 import com.daml.platform.server.api.ApiException
-import io.grpc.{Status, StatusRuntimeException}
-
+import com.google.protobuf.{Any => AnyProto}
+import com.google.rpc.Status
+import com.google.rpc.ErrorInfo
+import io.grpc.Status.Code
+import io.grpc.StatusRuntimeException
+import io.grpc.protobuf.StatusProto
 import scalaz.syntax.tag._
 
 trait ErrorFactories {
 
+  import ErrorFactories._
+
+  def notFound(message: String): StatusRuntimeException =
+    grpcError(
+      Status
+        .newBuilder()
+        .setCode(Code.NOT_FOUND.value())
+        .setMessage(message)
+        .addDetails(IndefiniteAnswerInfo)
+        .build()
+    )
+
   def ledgerIdMismatch(expected: LedgerId, received: LedgerId): StatusRuntimeException =
     grpcError(
-      Status.NOT_FOUND.withDescription(
-        s"Ledger ID '${received.unwrap}' not found. Actual Ledger ID is '${expected.unwrap}'."
-      )
+      Status
+        .newBuilder()
+        .setCode(Code.NOT_FOUND.value())
+        .setMessage(
+          s"Ledger ID '${received.unwrap}' not found. Actual Ledger ID is '${expected.unwrap}'."
+        )
+        .addDetails(IndefiniteAnswerInfo)
```

Comment:
Why is it indefinite? What's the source of this mapping? If this is going to be the authoritative source, I'd add comments that explain why.

Reply:
Good question. You reminded me that I should add more reviewers, as there is no source.

Reply:
That's an good question. `IndefiniteAnswerInfo` is certainly the safe choice here. We'd have to look at all use cases to understand whether we could make it definite here.